From aa37d517ef61ab7ab87ae7491dbe290a0c0566f3 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Jun 2021 16:56:17 +0300
Subject: [PATCH] [SW-46517] greco: update dccm address used for ppboot

In secureboot flow of greco, ppboot is copied into fw sram
and ppboot running on arc1 uses dccm for stack and data.
The current offset used for ppboot overlaps with dccm range
used by arcboot ROM which causes issues with ppboot execution.

This patch moves dccm address for ppboot so that it won't overlap
with arcboot ROM dccm useage.

Change-Id: I08ea87c8935dc48c5ee15b4517d5f1f042cb9904
---

diff --git a/arch/greco_arc/common.h b/arch/greco_arc/common.h
index 4ea11ab..ae92092 100644
--- a/arch/greco_arc/common.h
+++ b/arch/greco_arc/common.h
@@ -24,10 +24,10 @@
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
 #define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
- * Let PPBoot start from the offset of 8KB, for using DCCM
+ * Let PPBoot start from the offset of 8544 bytes, for using DCCM
  * for its own BSS and Stack.
  */
-#define PPBOOT_DCCM_BASE	0x80002000
+#define PPBOOT_DCCM_BASE	0x80002160
 
 #define SDRAM_BASE		0x10000000
 #define HL_SPI_BASE		0x1A000000
